apps/opik-frontend/src/components/pages/PromptsPage/AddEditPromptDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useState } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import usePromptCreateMutation from "@/api/prompts/usePromptCreateMutation";
+import { Prompt } from "@/types/prompts";
+import {
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+  Accordion,
+} from "@/components/ui/accordion";
+import usePromptUpdateMutation from "@/api/prompts/usePromptUpdateMutation";
+
+type AddPromptDialogProps = {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  prompt?: Prompt;
+};
+
+const AddEditPromptDialog: React.FunctionComponent<AddPromptDialogProps> = ({
+  open,
+  setOpen,
+  prompt: defaultPrompt,
+}) => {
+  const [name, setName] = useState(defaultPrompt?.name || "");
+  const [template, setTemplate] = useState("");
+  const [description, setDescription] = useState(
+    defaultPrompt?.description || "",
+  );
+
+  const promptCreateMutation = usePromptCreateMutation();
+  const promptUpdateMutation = usePromptUpdateMutation();
+
+  const isEdit = !!defaultPrompt;
+  const isValid = Boolean(name.length && (isEdit || template.length));
+
+  const createPrompt = useCallback(() => {
+    promptCreateMutation.mutate({
+      prompt: {
+        name,
+        template,
+        ...(description ? { description } : {}),
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [name, description, template, promptCreateMutation.mutate]);
```

Comment:
Maybe we should start destructuring the mutation like `const { mutate: createPrompt } = usePromptCreateMutation();` so we can use it as dependency here